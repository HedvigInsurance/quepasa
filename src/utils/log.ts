import { SegmentAnalyticsJs } from '../interfaces'

// tslint:disable no-console

const redBold = 'color: #ff0000; font-weight: bold'
const blueBold = 'color: #0000ff; font-weight: bold'
const blackBold = 'color: #000000; font-weight: bold'

export const wrapAnalyticsWithDebugLogging = (
  analytics: SegmentAnalyticsJs,
): SegmentAnalyticsJs => {
  const track: SegmentAnalyticsJs['track'] = (...args) => {
    console.group('%c[TRACK]', redBold)
    console.log(`event: %c${args[0]}`, blackBold)
    console.log('properties: ', args[1])
    console.log('options: ', args[2])
    console.groupEnd()
    analytics.track(...args)
  }
  const identify: SegmentAnalyticsJs['identify'] = (...args) => {
    console.log('%c[IDENTIFY]', blueBold)
    console.log(`userId: %c${args[0]}`, blackBold)
    console.log('traits: ', args[1])
    console.log('options: ', args[2])
    analytics.identify(...args)
  }
  return {
    track,
    identify,
  }
}
