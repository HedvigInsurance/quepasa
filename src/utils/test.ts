import { setupTrackers } from '..'
import { SegmentAnalyticsJs } from '../interfaces'

export enum TestEvents {
  Something = 'something',
  SomethingElse = 'something-else',
}

export const setup = () => {
  const mockAnalytics = makeMockAnalytics()
  const trackers = setupTrackers<TestEvents>(() => mockAnalytics)
  return {
    mockAnalytics,
    trackers,
  }
}

export const makeMockAnalytics = () => ({
  identify: jest.fn<SegmentAnalyticsJs['identify']>(),
  track: jest.fn<SegmentAnalyticsJs['track']>(),
})
