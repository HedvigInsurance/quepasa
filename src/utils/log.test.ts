import { wrapAnalyticsWithDebugLogging } from './log'
import { makeMockAnalytics } from './test'

// tslint:disable no-console

it('Matches snapshot for track', () => {
  const spy = jest.spyOn(console, 'log')

  const debugLogAnalytics = wrapAnalyticsWithDebugLogging(makeMockAnalytics())
  debugLogAnalytics.track('foo')
  expect(spy.mock.calls[0]).toMatchSnapshot()
  jest.clearAllMocks()
})

it('Matches snapshot for identify', () => {
  const spy = jest.spyOn(console, 'log')

  const debugLogAnalytics = wrapAnalyticsWithDebugLogging(makeMockAnalytics())
  debugLogAnalytics.identify('1')
  expect(spy.mock.calls[0]).toMatchSnapshot()
  jest.clearAllMocks()
})
