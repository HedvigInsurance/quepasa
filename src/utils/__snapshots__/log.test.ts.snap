// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot for identify 1`] = `
Array [
  "%c[IDENTIFY]",
  "color: #0000ff; font-weight: bold",
]
`;

exports[`Matches snapshot for track 1`] = `
Array [
  "event: %cfoo",
  "color: #000000; font-weight: bold",
]
`;
